### PR TITLE
Kazoo 3089

### PR DESCRIPTION
**Pull request by jamesaimonetti**
Fax to email

### CODE REVIEW

File: applications/callflow/src/module/cf_voicemail.erl [hunk 1]
```diff
@@ -305,6 +305,7 @@ compose_voicemail(#mailbox{max_message_count=MaxCount
                                                andalso MaxCount > 0 ->
     lager:debug("voicemail box is full, cannot hold more messages, sending notification"),
     Props = [{<<"Account-DB">>, whapps_call:account_db(Call)}
+             ,{<<"Account-ID">>, whapps_call:account_id(Call)}
```
Comment (macpie) [inline applications/callflow/src/module/cf_voicemail.erl]: Why not only put <<"Account-ID">> and use wh_util:format_account_id (or the other way around) ?

File: applications/teletype/src/teletype_tests.erl [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+-export([voicemail_to_email/1, voicemail_to_email/2
+         ,voicemail_full/1, voicemail_full/2
+         ,fax_inbound_to_email/1, fax_inbound_to_email/2
+         ,skel/1, skel/2
+        ]).
+
+-include("teletype.hrl").
+
+voicemail_to_email(AccountId) ->
```
Comment (macpie) [inline applications/teletype/src/teletype_tests.erl]: specs

File: applications/teletype/src/teletype_tests.erl [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    case find_vmboxes(AccountId) of
+        [] -> lager:debug("no voicemail boxes in account ~s", [AccountId]);
+        VMBoxes -> find_vmbox_with_messages(AccountId, VMBoxes)
+    end.
+
+find_vmboxes(AccountId) ->
```
Comment (macpie) [inline applications/teletype/src/teletype_tests.erl]: specs everywhere !

File: applications/teletype/src/templates/teletype_template_skel.erl [hunk 4]
```diff
@@ -65,37 +75,67 @@ handle_req(JObj, _Props) ->
...
+-spec build_macro_data(wh_json:object()) -> wh_proplist().
+build_macro_data(DataJObj) ->
+    wh_json:foldl(fun(MacroKey, _V, Acc) ->
+                          maybe_add_macro_key(MacroKey, Acc, DataJObj)
+                  end
+                  ,[]
+                  ,?TEMPLATE_MACROS
+                 ).
+
+set_to_address(DataJObj) ->
```
Comment (macpie) [inline applications/teletype/src/templates/teletype_template_skel.erl]: spec

File: applications/teletype/src/templates/teletype_template_skel.erl [hunk 5]
```diff
@@ -65,37 +75,67 @@ handle_req(JObj, _Props) ->
...
+    AccountId = wh_json:get_value(<<"account_id">>, DataJObj),
+    UserId = wh_json:get_value(<<"user_id">>, DataJObj),
+    {'ok', UserJObj} = couch_mgr:open_cache_doc(wh_util:format_account_id(AccountId, 'encoded')
+                                                ,UserId
+                                               ),
+    wh_json:set_value(<<"to">>, [find_email(UserJObj)], DataJObj).
+
+find_email(UserJObj) ->
```
Comment (macpie) [inline applications/teletype/src/templates/teletype_template_skel.erl]: spec

File: core/whistle_config-1.0.0/src/whapps_config.erl [hunk 6]
```diff
@@ -364,7 +371,7 @@ maybe_save_category(Category, JObj) ->
     maybe_save_category(Category, JObj, 'false').
 
 maybe_save_category(Category, JObj, Looped) ->
```
Comment (macpie) [inline core/whistle_config-1.0.0/src/whapps_config.erl]: spec

File: applications/callflow/src/module/cf_voicemail.erl (hunk 1 above)
Comment (jamesaimonetti) [inline applications/callflow/src/module/cf_voicemail.erl]: No need to do the format when we have both values computed in the call record.

File: core/whistle_config-1.0.0/src/whapps_config.erl (hunk 6 above)
Comment (jamesaimonetti) [inline core/whistle_config-1.0.0/src/whapps_config.erl]: It is above `maybe_save_category/2`

**Pull request merged**